### PR TITLE
Test Data compression and password protection

### PR DESCRIPTION
**Pull request by asamolion**

### CODE REVIEW

File: openedx/features/data_extract/helpers.py [hunk 1]
```diff
@@ -78,7 +78,13 @@ def get_user_demographic_data(profile):
                              data={'_uid': 1, 'username': profile.user.username},
                              headers=headers)
 
-    user_community_data = json.loads(response._content)['payload']
+    try:
+        user_community_data = json.loads(response._content)['payload']
+        reputation = user_community_data['reputation']
```
Comment (muhammadnadeem) [inline openedx/features/data_extract/helpers.py]: you can use user_community_data.get('reputation', 0) to skip exception handling

File: openedx/features/data_extract/management/commands/get_course_analytics.py [hunk 2]
```diff
@@ -49,17 +51,14 @@ def handle(self, **options):
                     'progress_data': progress_data,
                 })
 
-            with tempfile.TemporaryFile() as tmp:
+            with tempfile.NamedTemporaryFile() as tmp:
+                pyminizip.compress(tmp.name, '/tmp/data.zip', 'philu2miriam', 1)
```
Comment (muhammadnadeem) [inline openedx/features/data_extract/management/commands/get_course_analytics.py]: for now its ok, you should take password from user in future

File: openedx/features/data_extract/helpers.py (hunk 1 above)
Comment (asamolion) [reply openedx/features/data_extract/helpers.py]: done

File: openedx/features/data_extract/management/commands/get_course_analytics.py (hunk 2 above)
Comment (asamolion) [reply openedx/features/data_extract/management/commands/get_course_analytics.py]: will do

**Pull request merged**